controllers/lookup/druidlookup_controller.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lookup
+
+import (
+	"context"
+	druidv1alpha1 "github.com/datainfrahq/druid-operator/apis/druid/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"time"
+)
+
+// DruidLookupReconciler reconciles a DruidLookup object
+type DruidLookupReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=druid.apache.org,resources=druidlookups,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=druid.apache.org,resources=druidlookups/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=druid.apache.org,resources=druidlookups/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.4/pkg/reconcile
+func (r *DruidLookupReconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result, error) {
+	logr := log.FromContext(ctx)
+
+	reports := make(map[types.NamespacedName]Report)
+	k8sClient := NewK8sClient(r.Client)
+
+	lookupSpecsPerCluster, err := k8sClient.FindLookups(ctx, reports)
```

Comment:
As I mentioned earlier in this review.
It's better if you change this to:

`lookupSpecsPerCluster, err := r.FindLookups(ctx, reports)` and have this function under `DruidLookupReconciler`.

Another option is to make this function static and pass the client as a parameter (this is what we usually do in the controller). example: `lookupSpecsPerCluster, err := findLookups(ctx, client, reports)`

Reply:
Became as your first suggestion when removing `K8sClient` in c6b23dd.